providers/capa/upgrade/test_data/cluster_values.yaml

```diff
@@ -1 +1,17 @@
 # Values provided here merge on top of the default values found in https://github.com/giantswarm/cluster-standup-teardown
+global:
+  nodePools:
+    # We are using a name with 10 chars which is the max number of characters allowed by our kyverno policies.
+    #
+    # Here we override the default values from cluster-standup-teardown and increase min size from 2 to 3 because of
+    # issues discovered on 14/06/2024, where after the upgrade cluster-autoscaler scales down the cluster to min size
+    # (which is 2 by default) and after that a lot of Pods cannot be scheduled. The result is that Upgrade suite fails
+    # because WC Deployments are not ready, as test time-out passes more quickly than cluster-autoscaler can scale up
+    # the cluster.
+    nodepool-0:
+      maxSize: 5
+      minSize: 3
```

Comment:
It should be enough to only provide this property if that's the only one we're changing. The rest should be merged together with the base values I think.

Reply:
I feel like this is not even the root cause. I've seen clusters with 2 nodes working, but sometimes Cluster Autoscaler itself cannot be scheduled which then leads to the cluster not getting new nodes to overcome unschedulable pods.

Reply:
> I feel like this is not even the root cause.

IMO the number of nodes is definitely not the root cause of the issues we're seeing. Increasing the min size here is just a mitigation.

> but sometimes Cluster Autoscaler itself cannot be scheduled which then leads to the cluster not getting new nodes to overcome unschedulable pods

In the test failures I was debugging, after the upgrade cluster-autoscaler scaled down the cluster from 4-5 worker nodes to min 2 worker nodes. After the cluster was scaled down the Pods could not have been scheduled anymore and the test (checking if the Deployments were ready) timed out.

Reply:
The mitigation here is to prevent it to scale down too much.

Reply:
Even crazier, in another test run for this P (so with 3 min nodes), the cluster scaled down to just 1 node after the upgrade, see here https://github.com/giantswarm/cluster-test-suites/pull/342#issuecomment-2171965176.

Reply:
Here is a smaller PR with just node pool min size bump https://github.com/giantswarm/cluster-test-suites/pull/344